File: ExamenJava/Programme.java
import java.util.Scanner;

public class Programme {
    private final String PASSWORD = "pass1234*";
    private Scanner sc = new Scanner(System.in);
    private Zoo zoo = new Zoo();


    private boolean validPassword = false;
    public void runProgramme(){   
        //zoo.prueba();//MÉTODO PARA HACER PRUEBAS
        while (!validPassword) {
            System.out.println("\n******************** WELCOME ******************** ");
            System.out.println("Please insert the password to access to the zoo information");
            System.out.println("PASSWORD: pass1234* ");
            String cad = sc.nextLine();
                if(cad.equals(PASSWORD)){
                    validPassword = true;
                    showOptions();
                }
                else{
                    System.out.println("Wrong password. Try again");
                } 
        }
    }

    public void showOptions(){
    boolean flag = true;
        while (flag) {
            System.out.println("\n************** WELCOME TO THE ZOO SYSTEM **************");
            System.out.println("Please select an option(Write the number)");
            System.out.println("1. Register");
            System.out.println("2. Delete");
            System.out.println("3. Modify information");
            System.out.println("4. Consult information");
            System.out.println("5. Exit of the system");

            int x = sc.nextInt();
            switch (x) {
                case 1:
                System.out.println("What do you want to register? ");
                    System.out.println("Select an option");
                    System.out.println("1. Register an employee");
                    System.out.println("2. Register a visitant");
                    System.out.println("3. Register an animal");
                    System.out.println("4. Register a zoo visit");
                    System.out.println("5. Register the maintenance done to an animal");
                    System.out.println("6. Get back to the last menu");

                        switch (sc.nextInt()) {
                            case 1:
                                zoo.addEmployee();
                            break;

                            case 2:
                                zoo.addVisitant();
                            break;

                            case 3:
                                zoo.addAnimal();
                            break;

                            case 4:
                                zoo.addZooVisit();
                            break;

                            case 5:
                                zoo.addMaintenance();
                            break;
                        
                            default:
                            break;
                        }
                break;

                case 2:
                    System.out.println("1. Delete an employee");
                    System.out.println("2. Delete a visitant");
                    System.out.println("3. Delete an animal");
                    System.out.println("4. Get back to the last menu");

                        switch (sc.nextInt()) {
                            case 1:
                                zoo.removeEmployee();
                            break;

                            case 2:
                                zoo.removeVisitant();
                            break;

                            case 3:
                                zoo.removeAnimal();
                            break;
                        
                            default:
                            break;
                        }
                break;

                case 3:
                    System.out.println("1. Modify employee information");
                    System.out.println("2. Modify visitant information");
                    System.out.println("3. Modify animal information");
                    System.out.println("4. Get back to the last menu");

                        switch (sc.nextInt()) {
                            case 1:
                                zoo.modifyEmployee();
                            break;

                            case 2:
                                zoo.modifyVisitant();
                            break;

                            case 3:
                                zoo.modifyAnimal();
                            break;
                        
                            default:
                            break;
                        }
                break;

                case 4: 
                    System.out.println("1. Employees information");
                    System.out.println("2. Visitants information");
                    System.out.println("3. Animals information");
                    System.out.println("4. Zoo Visits information");
                    System.out.println("5. Maintenances information");
                    System.out.println("6. Get back to the last menu");


                        switch (sc.nextInt()) {
                            case 1:
                                System.out.println("Do you need information of a specific employee?");
                                System.out.println("1. Yes");
                                System.out.println("2. No");
                                int option1 = sc.nextInt();
                                    if(option1 == 1){
                                        zoo.showEmployeeInfo();
                                    }
                                    else if(option1 == 2){
                                        zoo.showEmployeeList();
                                    }
                                    else{
                                        System.out.println("Invalid option");
                                    }
                            break;

                            case 2:
                                System.out.println("Do you need information of a specific visitant?");
                                System.out.println("1. Yes");
                                System.out.println("2. No");
                                    int option2 = sc.nextInt();
                                    if(option2 == 1){
                                        zoo.showVisitantInfo();
                                    }
                                    else if(option2 == 2){
                                        zoo.showVisitantList();
                                    }
                                    else{
                                        System.out.println("Invalid option");
                                    }
                            break;

                            case 3:
                                System.out.println("Do you need information of a specific animal?");
                                System.out.println("1. Yes");
                                System.out.println("2. No");
                                    int option3 = sc.nextInt();
                                    if(option3 == 1){
                                        zoo.showAnimalInfo();
                                    }
                                    else if(option3 == 2){
                                        zoo.showAnimalList();
                                    }
                                    else{
                                        System.out.println("Invalid option");
                                    }
                            break;

                            case 4:
                                System.out.println("Do you need information of a specific zoo visit?");
                                System.out.println("1. Yes");
                                System.out.println("2. No");
                                    int option4 = sc.nextInt();
                                    if(option4 == 1){
                                        zoo.showZooVisitInfo();
                                    }
                                    else if(option4 == 2){
                                        zoo.showZooVisitList();
                                    }
                                    else{
                                        System.out.println("Invalid option");
                                    }
                            break;

                            case 5:
                                System.out.println("Do you need information of a specific maintenance?");
                                System.out.println("1. Yes");
                                System.out.println("2. No");
                                int option5 = sc.nextInt();
                                    if(option5 == 1){
                                        zoo.showMaintenanceInfo();
                                    }
                                    else if(option5 == 2){
                                        zoo.showMaintenanceList();
                                    }
                                    else{
                                        System.out.println("Invalid option");
                                    }
                            break;
                        
                            default:
                            break;
                        }
                break;

                default:
                    flag = false;
                break;
            }
        }
    }
}

File: ExamenPy/Programme.py
from Zoo import Zoo
from ZooVisit import ZooVisit
from Visitant import Visitant
from Employee import Employee
from Maintenance import Maintenance

class Programme:
    __PASSWORD = "pass1234*"
    zoo = Zoo()
    
    def run_programme(self):
        valid_password = False
        while not valid_password:
            print("******************** WELCOME ********************")
            print("Please insert the password to access the zoo information")
            print("PASSWORD: pass1234*")
            password = input()
            if password == self.__PASSWORD:
                valid_password = True
                self.show_options()
            else:
                print("Wrong password. Try again")
    
    def show_options(self):
        flag = True
        while flag:
            print("************** WELCOME TO THE ZOO SYSTEM **************")
            print("Please select an option (Write the number)")
            print("1. Register")
            print("2. Delete")
            print("3. Modify information")
            print("4. Consult information")
            print("5. Exit of the system")
            
            x = int(input())
            if x == 1:
                print("What do you want to register?")
                print("Select an option")
                print("1. Register an employee")
                print("2. Register a visitant")
                print("3. Register an animal")
                print("4. Register a zoo visit")
                print("5. Register the maintenance done to an animal")
                print("6. Get back to the last menu")
                
                option = int(input())
                if option == 1:
                    self.zoo.add_employee()
                elif option == 2:
                    self.zoo.add_visitant()
                elif option == 3:
                    self.zoo.add_animal()
                elif option == 4:
                    self.zoo.add_zoo_visit()
                elif option == 5:
                    self.zoo.add_maintenance()
                else:
                    pass
                
            elif x == 2:
                print("1. Delete an employee")
                print("2. Delete a visitant")
                print("3. Delete an animal")
                print("4. Get back to the last menu")
                
                option = int(input())
                if option == 1:
                    self.zoo.remove_employee()
                elif option == 2:
                    self.zoo.remove_visitant()
                elif option == 3:
                    self.zoo.remove_animal()
                else:
                    pass
                
            elif x == 3:
                print("1. Modify employee information")
                print("2. Modify visitant information")
                print("3. Modify animal information")
                print("4. Get back to the last menu")
                
                option = int(input())
                if option == 1:
                    self.zoo.modify_employee()
                elif option == 2:
                    self.zoo.modify_visitant()
                elif option == 3:
                    self.zoo.modify_animal()
                else:
                    pass
                
            elif x == 4:
                print("1. Employees information")
                print("2. Visitants information")
                print("3. Animals information")
                print("4. Zoo Visits information")
                print("5. Maintenances information")
                print("6. Get back to the last menu")
                
                option = int(input())
                if option == 1:
                    print("Do you need information of a specific employee?")
                    print("1. Yes")
                    print("2. No")
                    sub_option = int(input())
                    if sub_option == 1:
                        self.zoo.show_employee_info()
                    elif sub_option == 2:
                        self.zoo.show_employee_list()
                    else:
                        print("Invalid option")
                elif option == 2:
                    print("Do you need information of a specific visitant?")
                    print("1. Yes")
                    print("2. No")
                    sub_option = int(input())
                    if sub_option == 1:
                        self.zoo.show_visitant_info()
                    elif sub_option == 2:
                        self.zoo.show_visitant_list()
                    else:
                        print("Invalid option")
                elif option == 3:
                    print("Do you need information of a specific animal?")
                    print("1. Yes")
                    print("2. No")
                    sub_option = int(input())
                    if sub_option == 1:
                        self.zoo.show_animal_info()
                    elif sub_option == 2:
                        self.zoo.show_animal_list()
                    else:
                        print("Invalid option")
                elif option == 4:
                    print("Do you need information of a specific zoo visit?")
                    print("1. Yes")
                    print("2. No")
                    sub_option = int(input())
                    if sub_option == 1:
                        self.zoo.show_zoo_visit_info()
                    elif sub_option == 2:
                        self.zoo.show_zoo_visit_list()
                    else:
                        print("Invalid option")
                elif option == 5:
                    print("Do you need information of a specific maintenance?")
                    print("1. Yes")
                    print("2. No")
                    sub_option = int(input())
                    if sub_option == 1:
                        self.zoo.show_maintenance_info()
                    elif sub_option == 2:
                        self.zoo.show_maintenance_list()
                    else:
                        print("Invalid option")
                else:
                    pass
            else:
                flag = False
